eventsources/eventing.go

```diff
@@ -402,6 +467,24 @@ func (e *EventSourceAdaptor) run(ctx context.Context, servers map[apicommon.Even
 				}
 				if err = common.Connect(&backoff, func() error {
 					return s.StartListening(ctx, func(data []byte, opts ...eventsourcecommon.Options) error {
+						if filter, ok := filters[s.GetEventName()]; ok {
+							dataMap := make(map[string]interface{})
```

Comment:
In terms of functionality, everything looks good to me.

Some minor suggestions - move these entire section into a function named `filterEvent(data []byte, filter *EventSourceFilter) (bool, error)`, which includes the content of `shouldDispatch()`;